orttraining/orttraining/core/graph/optimizer/adam_optimizer_builder.cc

```diff
@@ -52,8 +75,21 @@ Status AdamOptimizerBuilder::Build(
     // In distributed training, some weights may not be updated by all ranks.
     if (opt_configs[i].enabled) {
       // The type proto initializer for Update Count
-      const std::string update_count_string = "Update_Count_" + weight_name;  // per weight optimizer requires a per weight update count
-      TensorProto uc_tensor_proto = CreateTensorProto<int64_t>(update_count_string, 1);
+      const std::string uc_prefix = "Update_Count";
+      const std::string update_count_string = uc_prefix + "_" + weight_name;  // per weight optimizer requires a per weight update count
+      TensorProto uc_tensor_proto;
+
+      // Update 'Update_Count' initializer with init value
+      const auto& initial_states = opt_configs[i].initial_states;
+      const auto uc_state_it = initial_states.find(uc_prefix);
```

Comment:
Do you not need to index into the map with the weight name? The structure is 
```
model_weight_name:
{
    "Moment_1": moment1_tensor,
    "Moment_2": moment2_tensor,
    "Update_Count": update_tensor # if optimizer is adam, absent otherwise
},
"shared_optimizer_state": # if optimizer is shared, absent otherwise. So far, only lamb optimizer uses this.
{
    "step": step_tensor # int array of size 1
}
```

So, the ```"Update_Count"``` should be one level deeper in the map. Please correct me if my understanding is incorrect.

Reply:
the map that is accessed here is already set per weight(done in training_session.cc), meaning this is not required.